src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java

```
@@ -293,6 +294,48 @@ public Response downloadZip(JsonObject content,
       }
    }
 
+   @POST
+   @javax.ws.rs.Path("/commands/{commandName}/catapult")
+   @Consumes(MediaType.APPLICATION_JSON)
+   public Response uploadZip(JsonObject content,
+                               @PathParam("commandName") @DefaultValue(DEFAULT_COMMAND_NAME) String commandName,
+                               @Context HttpHeaders headers)
+         throws Exception
+   {
+      Response response = downloadZip(content, commandName, headers);
+      if (response.getStatus() != 200) {
+         return response;
+      }
+
+      String fileName = findArtifactId(content);
+      byte[] zipContents = (byte[]) response.getEntity();
+
+      UriBuilder uri = UriBuilder.fromPath("/api/catapult/upload");
+      uri.host(System.getenv("CATAPULT_SERVICE_HOST"));
```

Perhaps we should specify a default value or throw an error if that ENV is not specified?

---

don't know if it always there if you deploy first the generator backend end then catapult. Don't want this service to explode if catapult is not there. Not sure what the best default action would be

---

In case Catapult support is "optional", I'd suggest to introduce something like `WITH_CATAPULT` env var.  Based on that, check for `CATAPULT_SERVICE_HOST` and fail if `WITH_CATAPULT == True` and `CATAPULT_SERVICE_HOST` is undefined - that would make sure to give you proper "ordering" in OpenShift.

---

Or `CATAPULT_ENABLED` as the env var name 😄